axelrod/result_set.py

```diff
@@ -45,6 +47,8 @@ def __init__(self, players, interactions, progress_bar=True):
             progress_bar : bool
                 Whether or not to create a progress bar which will be updated
         """
+        if game is not None:
+            self.game = game
```

Comment:
I don't think that would do the trick actually.

However, what could be nicer than current is to have:

```
__init__(self, players, interactions, progress_bar=True, game=Game()):
```

and then:

```
self.game = game
```

The current setup mimics the tournament class setup which sets the class attribute to be the default game and then changes it if a different one is passed in init.